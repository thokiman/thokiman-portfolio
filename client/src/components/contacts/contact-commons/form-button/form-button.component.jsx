import React from 'react';
import { FormButtonContainer } from './form-button.styles';

export const FormButton = ({ children, ...props }) => {
  return (
    <FormButtonContainer
      className='contact-form-element'
      type='button'
      {...props}
    >
      {children}
    </FormButtonContainer>
  );
};

export default FormButton;
