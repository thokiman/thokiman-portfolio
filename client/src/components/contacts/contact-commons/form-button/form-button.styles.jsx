import styled from 'styled-components';

const heavyBlueLightColor = '#bccfd5';
const backgroundColorFormButtonContainer = 'rgba(63, 66, 73, 0.7)';
const greenBlueColor = '#128194';
export const FormButtonContainer = styled.button`
  min-width: 165px;
  width: auto;
  height: 50px;
  letter-spacing: 0.5px;
  line-height: 50px;
  padding: 0 35px 0 35px;
  font-size: 35px;
  background-color: ${backgroundColorFormButtonContainer};
  color: ${heavyBlueLightColor};
  border: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  font-family: futura-medium;
  &:hover {
    transition: all ease-in-out 0.3s;
    background-color: ${greenBlueColor};
    color: ${heavyBlueLightColor};
  }
`;

FormButtonContainer.displayName = 'FormButtonContainer';
