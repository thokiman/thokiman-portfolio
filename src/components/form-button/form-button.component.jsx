import React from "react";

import "./form-button.style.scss";

export const FormButton = ({ children, ...props }) => {
  return (
    <button className="form-button" type="button" {...props}>
      {children}
    </button>
  );
};

export default FormButton;
