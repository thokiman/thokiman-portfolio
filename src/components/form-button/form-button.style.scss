.form-button {
  min-width: 165px;
  width: auto;
  height: 50px;
  letter-spacing: 0.5px;
  line-height: 50px;
  padding: 0 35px 0 35px;
  font-size: 35px;
  background-color: rgba($color:#3f4249, $alpha: 0.7);
  color: #bccfd5;

  border: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  font-family: futura-medium;

  &:hover {
    transition: all ease-in-out 0.3s;
    background-color: rgba($color: #128194, $alpha: 1);
    color: #bccfd5;

  }

}