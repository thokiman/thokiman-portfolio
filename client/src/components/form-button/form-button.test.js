import React from "react";
import { shallow } from "enzyme";
import { FormButton } from "./form-button.component";

it("should render FormButton component", () => {
  expect(shallow(<FormButton />)).toMatchSnapshot();
});
