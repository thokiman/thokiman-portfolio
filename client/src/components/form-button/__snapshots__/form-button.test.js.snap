// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormButton component 1`] = `ShallowWrapper {}`;
